### PR TITLE
Non exported error codes

### PR DESCRIPTION
**Pull request by aricart**
Several error codes were not exported to typescript.

### CODE REVIEW

**Comment by aricart** [comment]
> That is not proper versioning is it or is this just temporary?

just temporary, when we release will drop the `-`. This way we can maintain a future release stream at `nats@next`.  See https://medium.com/@albertoricart/simple-npm-release-workflow-71e4113f8272.

**Comment by derekcollison** [comment]
ok gotcha.

File: tstest/main.ts [hunk 1]
```diff
@@ -12,8 +12,31 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-import {connect} from '..'
+import {
+    connect,
+    BAD_AUTHENTICATION,
```
Comment (derekcollison) [inline tstest/main.ts]: Should we put these under something so we can pull them all in with one import?

File: tstest/main.ts [hunk 2]
```diff
@@ -121,3 +144,41 @@ roid = nc.requestOne('bar', 'payload', {max: 10}, 100, (payload:string) => {
...
+    SUB_DRAINING
+];
+
+codes.forEach((v) => {
+    if(v.length === 0) {
+        throw new Error("there's a problem with a constant");
+    } else {
+        console.log(v);
+    }
+});
```
Comment (derekcollison) [inline tstest/main.ts]: Add NL

File: tstest/main.ts (hunk 1 above)
Comment (aricart) [reply tstest/main.ts]: for typescript each export is imported.

File: tstest/main.ts (hunk 2 above)
Comment (aricart) [reply tstest/main.ts]: done

File: tstest/main.ts (hunk 1 above)
Comment (derekcollison) [reply tstest/main.ts]: But could we import "CODES" and do CODES.<SOMETHING>?

Comment (derekcollison) [reply tstest/main.ts]: ok I see this would have to be a 2.0 thing..

**Comment by aricart** [comment]
> LGTM _ let's group these for 2.0.

yeap.

**Pull request merged**